### ISSUE TITLE
Can't pass through large responses

### ISSUE THREAD

**Issue by mayrand** [ThreeMammals/Ocelot#881]
## Expected Behavior / New Feature

Implement option to pass through response without loading it to memory to be able to work with large responses.

## Actual Behavior / Motivation for New Feature

Response limit is 2GB.

## My quick solution
```
src/Ocelot/Requester/HttpClientWrapper.cs
@@ -17,7 +17,7 @@ namespace Ocelot.Requester
 
         public Task<HttpResponseMessage> SendAsync(HttpRequestMessage request)
         {
-            return Client.SendAsync(request);
+            return Client.SendAsync(request, HttpCompletionOption.ResponseHeadersRead);
         }
     }
 }

```

**Comment by ztzsws1** [comment]
![image](https://user-images.githubusercontent.com/5999059/59586690-12aac400-9116-11e9-8d0b-981cf48d3a83.png)
I have same issue, can not download file which larger that 2G. 
If I change ocelot code like @mayrand , it works fine. I am not sure if this will cause other problems.
Is there any suggestion?

**Comment by ztzsws1** [comment]
Can we support to config which route use the option ResponseHeadersRead, and which one used option Default?
Maybe like this,
`{
      "DownstreamPathTemplate": "/api/{everything}",
      "DownstreamScheme": "http",
      "DownstreamHostAndPorts": [
        {
          "Host": "localhost",
          "Port": 12345
        }
      ],
      "UpstreamPathTemplate": "/common/api/{everything}",
      "UpstreamHttpMethod": [ "Get", "Post", "Put", "Delete" ],
      "HttpCompletionOption":"ResponseHeadersRead" --------------> Here
    },`

**Comment by raman-m** [comment]
Hi @mayrand and @ztzsws1 !
Downloads & uploads of large files should be fixed in the current upcoming release, right @ggnaegi ?
Please watch for next version 23.0 and [the release milestone](https://github.com/ThreeMammals/Ocelot/milestone/2) will be closed soon...

**Comment by raman-m** [comment]
Fixed by #1724 and #1824
